src/ipc/ipc4/helper.c

```diff
@@ -1031,11 +1032,35 @@ int ipc4_find_dma_config(struct ipc_config_dai *dai, uint8_t *data_buffer, uint3
 	if (*dma_config_id != GTW_DMA_CONFIG_ID)
 		return IPC4_INVALID_REQUEST;
 
-	dai->host_dma_config = GET_IPC_DMA_CONFIG(data_buffer, size);
+	dai->host_dma_config[0] = GET_IPC_DMA_CONFIG(data_buffer, size);
 #endif
 	return IPC4_SUCCESS;
 }
 
+int ipc4_find_dma_config_multiple(struct ipc_config_dai *dai, uint8_t *data_buffer,
+				  uint32_t size, uint32_t device_id, int dma_cfg_idx)
+{
+	uint32_t end_addr = (uint32_t)data_buffer + size;
+	struct ipc_dma_config *dma_cfg;
+	struct sof_tlv *tlvs;
+
+	for (tlvs = (struct sof_tlv *)data_buffer; (uint32_t)tlvs < end_addr;
+	     tlvs = tlv_next(tlvs)) {
+		dma_cfg = tlv_value_ptr_get(tlvs, GTW_DMA_CONFIG_ID);
+		if (!dma_cfg)
+			continue;
+
+		for (uint32_t i = 0; i < dma_cfg->channel_map.device_count; i++) {
+			if (dma_cfg->channel_map.map[i].device_address == device_id) {
+				dai->host_dma_config[dma_cfg_idx] = dma_cfg;
+				return IPC4_SUCCESS;
+			}
+		}
```

Comment:
I can't use `tlv_value_get()` because it is able to find only the first tlv with the same type, but in case of aggregation there  will be two or more. Therefore, I will need to call `tlv_value_get()` in a loop which doesn't make any sense.